tools/test/toolchains/test_toolchains.py

```diff
@@ -30,6 +31,8 @@ def test_arm_minimal_printf(self):
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
         mock_target.printf_lib = "minimal-printf"
+        mock_target.default_lib = "std"
```

Comment:
```suggestion
        mock_target.default_lib = "sTd"
```
To cover case insensitivity. I would do this for all test cases (for all toolchains) where you do not expect an exception. I would not do it where you expect an exception.

Reply:
This test case is not testing default_lib. we have a test case to confirm case-insensitive works but as per our documentation, the default_lib options are a small letter.

Reply:
Oh ok. The test case is for minimal-printf.